pkg/controller/gitopsconfig/controller.go

```diff
@@ -76,7 +77,29 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	}
 
 	// Watch for changes to primary resource GitOpsConfig
-	err = c.Watch(&source.Kind{Type: &gitopsv1alpha1.GitOpsConfig{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(
+		&source.Kind{Type: &gitopsv1alpha1.GitOpsConfig{}},
+		&handler.EnqueueRequestForObject{},
+		// TODO: once we update to sigs.k8s.io/controller-runtime >=0.2.0, use their
+		// .../pkg/predicate.GenerationChangedPredicate instead of rewriting it on our own
+		predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				// If there's a status update, .metadata.Generation field isn't changed - ignore such event
+				if e.MetaOld.GetGeneration() == e.MetaNew.GetGeneration() {
+					return false
+				}
+				if e.MetaOld == nil {
+					log.Error(nil, "Update event has no old metadata", "event", e)
+					return false
+				}
+				if e.MetaNew == nil {
+					log.Error(nil, "Update event has no new metadata", "event", e)
+					return false
+				}
+				return true
```

Comment:
Maybe the return could be simplified by writing something like ...

```go
return e.MetaOld.GetGeneration() == e.MetaNew.GetGeneration()
```

This would eliminate the need for the conditional on lines 88 to 90.